### ISSUE TITLE
Plugin breaks checkout if streetname / housenumber can not be resolved

### ISSUE THREAD

**Issue by patchee500** [ozean12/shopware6-module#12]
It should be possible to reproduce this with Shopware 6.5.8.1 and Plugin version 3.0.0.

I have found that in cases where the Regex from \Billie\Sdk\Util\AddressHelper::STREETNUMBER_REGEX fails to identify the street of an address (e.g. when supplying "Teststraße 1, 3. OG" in the Shopware street field, null is returned and passed along as the street name), this will lead to a chrash in the checkout, effectively blocking  the customer from any further progress (see \Billie\BilliePayment\Components\BillieApi\Util\AddressHelper::createAddress, which is triggered by \Billie\BilliePayment\Components\Checkout\Subscriber\CheckoutSubscriber somewhere down the line).

While I understand supplying the street this way is not exactly DIN conforming, I don't think it should block the customer from any further progress in the Checkout.

I'm also unsure how important the house number really is for Billie, but seeing that it is flagged as optional in \Billie\Sdk\Model\Address I wonder if \Billie\BilliePayment\Components\BillieApi\Util\AddressHelper::createAddress can't be changed to skip the streetname/number splitting entirely as Shopware natively doesn't support street and house number as split fields:

```
    public static function createAddress(object $addressEntity): Address
    {
        self::validateParam($addressEntity);

        return (new Address())
            ->setValidateOnSet(false)
            ->setStreet($addressEntity->getStreet())
            ->setHouseNumber(null)
            ->setPostalCode($addressEntity->getZipcode())
            ->setCity($addressEntity->getCity())
            ->setCountryCode($addressEntity->getCountry()->getIso());
    }
```

**Comment by rommelfreddy** [comment]
Hello @patchee500 

thank you for your report!

as you suggested, this would be the easiest solution. 
We also implemented this in the next upcoming release. 

We also implemented a fix that the SDK does not validate during the page-load. 
https://github.com/ozean12/shopware6-payment-module/commit/9070b0905f673d0d45708fa51d5e98c521d2a592
Sadly an SDK update is required for this (this has been finally released last week - 3.1.x)

I will close this issue, because it has been fixed on two ways, which will be released within the upcoming release. 
If you still have issues with this after the release, feel free to reopen the issue.

**Comment by rommelfreddy** [comment]
Hello @patchee500 

thank you for your patience. just released the module with the latest changes of the last months :)
